fabric-loader-0.16.9服务端启动报错

java.lang.RuntimeException: Could not execute entrypoint stage 'main' due to errors, provided by 'quickbackupmulti' at 'io.github.skydynamic.quickbackupmulti.QuickBackupMulti'!
        at net.fabricmc.loader.impl.FabricLoaderImpl.lambda$invokeEntrypoints$2(FabricLoaderImpl.java:403) ~[fabric-loader-0.16.9.jar:?]
        at net.fabricmc.loader.impl.util.ExceptionUtil.gatherExceptions(ExceptionUtil.java:33) ~[fabric-loader-0.16.9.jar:?]
        at net.fabricmc.loader.impl.FabricLoaderImpl.invokeEntrypoints(FabricLoaderImpl.java:401) ~[fabric-loader-0.16.9.jar:?]
        at net.fabricmc.loader.impl.game.minecraft.Hooks.startServer(Hooks.java:63) ~[fabric-loader-0.16.9.jar:?]
        at knot/net.minecraft.server.Main.main(Main.java:111) [server-intermediary.jar:?]
        at net.fabricmc.loader.impl.game.minecraft.MinecraftGameProvider.launch(MinecraftGameProvider.java:480) [fabric-loader-0.16.9.jar:?]
        at net.fabricmc.loader.impl.launch.knot.Knot.launch(Knot.java:74) [fabric-loader-0.16.9.jar:?]
        at net.fabricmc.loader.impl.launch.knot.KnotServer.main(KnotServer.java:23) [fabric-loader-0.16.9.jar:?]
        at net.fabricmc.loader.impl.launch.server.FabricServerLauncher.main(FabricServerLauncher.java:69) [fabric-loader-0.16.9.jar:?]
        at net.fabricmc.installer.ServerLauncher.main(ServerLauncher.java:69) [fabric-server-mc.1.21-loader.0.16.9-launcher.1.0.1.jar:1.0.1]
Caused by: java.lang.NoClassDefFoundError: org/apache/http/client/HttpClient
        at knot/io.github.skydynamic.quickbackupmulti.QuickBackupMulti.<clinit>(QuickBackupMulti.java:52) ~[QuickBackupMulti-mc1.21-v2.2.1.jar:?]
        at java.base/java.lang.Class.forName0(Native Method) ~[?:?]
        at java.base/java.lang.Class.forName(Class.java:534) ~[?:?]
        at java.base/java.lang.Class.forName(Class.java:513) ~[?:?]
        at net.fabricmc.loader.impl.util.DefaultLanguageAdapter.create(DefaultLanguageAdapter.java:50) ~[fabric-loader-0.16.9.jar:?]
        at net.fabricmc.loader.impl.entrypoint.EntrypointStorage$NewEntry.getOrCreate(EntrypointStorage.java:124) ~[fabric-loader-0.16.9.jar:?]
        at net.fabricmc.loader.impl.entrypoint.EntrypointContainerImpl.getEntrypoint(EntrypointContainerImpl.java:53) ~[fabric-loader-0.16.9.jar:?]
        at net.fabricmc.loader.impl.FabricLoaderImpl.invokeEntrypoints(FabricLoaderImpl.java:399) ~[fabric-loader-0.16.9.jar:?]
        ... 7 more
Caused by: java.lang.ClassNotFoundException: org.apache.http.client.HttpClient
        at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:641) ~[?:?]
        at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:526) ~[?:?]
        at net.fabricmc.loader.impl.launch.knot.KnotClassDelegate.loadClass(KnotClassDelegate.java:226) ~[fabric-loader-0.16.9.jar:?]
        at net.fabricmc.loader.impl.launch.knot.KnotClassLoader.loadClass(KnotClassLoader.java:119) ~[fabric-loader-0.16.9.jar:?]
        at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:526) ~[?:?]
        at knot/io.github.skydynamic.quickbackupmulti.QuickBackupMulti.<clinit>(QuickBackupMulti.java:52) ~[QuickBackupMulti-mc1.21-v2.2.1.jar:?]
        at java.base/java.lang.Class.forName0(Native Method) ~[?:?]
        at java.base/java.lang.Class.forName(Class.java:534) ~[?:?]
        at java.base/java.lang.Class.forName(Class.java:513) ~[?:?]
        at net.fabricmc.loader.impl.util.DefaultLanguageAdapter.create(DefaultLanguageAdapter.java:50) ~[fabric-loader-0.16.9.jar:?]
        at net.fabricmc.loader.impl.entrypoint.EntrypointStorage$NewEntry.getOrCreate(EntrypointStorage.java:124) ~[fabric-loader-0.16.9.jar:?]
        at net.fabricmc.loader.impl.entrypoint.EntrypointContainerImpl.getEntrypoint(EntrypointContainerImpl.java:53) ~[fabric-loader-0.16.9.jar:?]
        at net.fabricmc.loader.impl.FabricLoaderImpl.invokeEntrypoints(FabricLoaderImpl.java:399) ~[fabric-loader-0.16.9.jar:?]
        ... 7 more

--- a/src/main/java/io/github/skydynamic/quickbackupmulti/utils/UpdateChecker.java
+++ b/src/main/java/io/github/skydynamic/quickbackupmulti/utils/UpdateChecker.java
@@ -3,17 +3,18 @@ package io.github.skydynamic.quickbackupmulti.utils;
 import com.google.gson.JsonObject;
 import com.google.gson.JsonParser;
 import io.github.skydynamic.quickbackupmulti.QuickBackupMulti;
-import org.apache.http.HttpResponse;
-import org.apache.http.client.HttpClient;
-import org.apache.http.client.methods.HttpGet;
-import org.apache.http.impl.client.HttpClients;
-import org.apache.http.util.EntityUtils;
+
 
 import java.io.IOException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
 
 public class UpdateChecker extends Thread {
 
-    private static final HttpClient CLIENT = HttpClients.createDefault();
+    private static final HttpClient CLIENT = HttpClient.newHttpClient();
     private static final String RELEASE_API_URL = "https://api.github.com/repos/QuickBackupMultiMod-Dev/QuickBackupM-Fabric/releases/latest";
 
     public String latestVersion;
@@ -31,11 +32,13 @@ public class UpdateChecker extends Thread {
                 QuickBackupMulti.LOGGER.warn("Current mod version is not found.");
                 return;
             }
-            HttpResponse httpResponse = CLIENT.execute(new HttpGet(RELEASE_API_URL));
-            String body = EntityUtils.toString(httpResponse.getEntity());
+            HttpResponse<String> response = CLIENT.send(
+                HttpRequest.newBuilder().uri(new URI(RELEASE_API_URL)).build(),
+                HttpResponse.BodyHandlers.ofString()
+            );
 
             // Get Meta data
-            JsonObject jsonObject = JsonParser.parseString(body).getAsJsonObject();
+            JsonObject jsonObject = JsonParser.parseString(response.body()).getAsJsonObject();
             latestVersion = jsonObject.get("tag_name").getAsString();
             latestVersionHtmUrl = jsonObject.get("html_url").getAsString();
 
@@ -52,8 +55,7 @@ public class UpdateChecker extends Thread {
                     "{} has new version {}. You can see: {}", QuickBackupMulti.modName, latestVersion, latestVersionHtmUrl
                 );
             }
-
-        } catch (IOException e) {
+        } catch (IOException | InterruptedException | URISyntaxException e) {
             QuickBackupMulti.LOGGER.error("Check update failed", e);
         }
     }

